internal/db/models/programs.marshal.go

```diff
@@ -0,0 +1,105 @@
+package models
+
+import (
+	"encoding/json"
+	"errors"
+
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+// I hate the below code but I spent waaay too much time on it to devise
+// a better solution...
+
+type RawElement []byte
+
+func (r RawElement) MarshalJSON() ([]byte, error) {
+	return json.RawMessage(r).MarshalJSON()
+}
+
+func (r *RawElement) UnmarshalJSON(data []byte) error {
+	return (*json.RawMessage)(r).UnmarshalJSON(data)
+}
+
+type MarshalFunc func(interface{}) ([]byte, error)
+type UnmarshalFunc func([]byte, interface{}) error
+
+func (ip *InteractionProgramInstructionList) marshal(Marshal MarshalFunc) ([]byte, error) {
+	typedInstructions := make([]interface{}, len(*ip))
+	for i, inst := range *ip {
+		switch inst := inst.(type) {
```

Comment:
Is there no Stop_Recording for marshall?

Reply:
It should be there; guess I didn't add it